security/advancedtls/crl.go

```diff
@@ -320,7 +320,7 @@ func parseCertIssuerExt(ext pkix.Extension) ([]byte, error) {
 	// GeneralNames ::= SEQUENCE SIZE (1..MAX) OF GeneralName
 	var generalNames []asn1.RawValue
 	if rest, err := asn1.Unmarshal(ext.Value, &generalNames); err != nil || len(rest) != 0 {
-		return nil, err
+		return nil, fmt.Errorf("asn1.Unmarshal failed err = %v", err)
```

Comment:
This isn't really what I had in mind.

This doesn't add any useful information that `Unmarshal` doesn't already include in its error, so it would be better to not wrap it than to add clutter.  What I meant was more about adding context of _why_ you were performing the operation that failed and interpreting what that means, usually only adding information that is not already directly passed to the function you just called.  E.g. 

```go
fmt.Errorf("error parsing <ASN.1 extension(??)> %q: %v", ext.Id /* ? */, err)
```

And where you call this function from:

```go
if ext.Critical {
	return RevocationUndetermined,
		fmt.Errorf("critical extension error for revoked certificate %v in x509 certificate %v: %v",
			revCert.SerialNumber.Text() /* ? */, c.SerialNumber.Text() /* ? */, err)
}
```

This way when an error happens it will be easier to track down exactly what caused it.

Reply:
Oops, I see now. I'll do another PR to correct the error message